cli/register/servicewait.go

```diff
@@ -24,115 +105,193 @@ func WaitForService(org string, waitService string, waitTimeout int, pattern str
 		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("--timeout must be a positive integer."))
 	}
 
+	servSpecArr := []serviceSpec{}
+	if pattern != "" {
+		for _, s := range pat.Services {
+			if (waitService == "*" || waitService == s.ServiceURL) && (org == "*" || org == s.ServiceOrg) && (s.ServiceArch == "*" || nodeArch == s.ServiceArch) {
+				if len(s.ServiceVersions) > 0 && CheckService(s.ServiceURL, s.ServiceOrg, nodeArch, s.ServiceVersions[0].Version, nodeType, userOrg, userPw) {
```

Comment:
Since CheckService only makes sure that the service type and node type matches, are you saying that different service version might have different service types?

Reply:
ah, my comment is invalid, you are correct. I made an incorrect assumption about why you are passing service version, and neglected to check what CheckService was actually doing. Please add a comment before line 112, something like: Collecting the set of services which apply to this node type and architecture.